crates/sui-types/src/execution_mode.rs

```diff
@@ -32,23 +33,19 @@ pub trait ExecutionMode {
 
     fn empty_results() -> Self::ExecutionResults;
 
-    fn add_argument_update<'vm, 'state, 'a, S: StorageView>(
-        context: &mut ExecutionContext<'vm, 'state, 'a, S>,
```

Comment:
`ExecutionContext` is part of the execution layer, so we can't depend on it here -- it's abstracted behind a trait.